packages/runtime/container-runtime/src/throttler.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Used to give increasing delay times for throttling a single functionality.
+ * Delay is based on previous attempts within specified time window, ignoring actual delay time.
+ */
+ export class Throttler {
+    private startTimes: number[] = [];
+    constructor(
+        private readonly delayWindowMs: number,
+        private readonly maxDelayMs: number,
+        private readonly delayFunction: (n: number) => number,
+    ) { }
+
+    public get attempts() {
```

Comment:
this looks wrong. won't the shift below affect the length and loose attempts?

Reply:
opened #6472 - to add testing & comments